src/ConstructiveSolidGeometry/ConstructiveSolidGeometry.jl

```diff
@@ -8,14 +8,17 @@ module ConstructiveSolidGeometry
 
     # # Other Packages
     using IntervalSets
+    using Parameters
+    using PolygonOps
```

Comment:
Can you explain what these packages are for?

Reply:
Parameters.jl is a nice package which is quite common nowadays. 
It gives you the macro `@with_kw`:
```julia
@with_kw struct Box{T, CO} <: AbstractVolumePrimitive{T, CO}
    hX::T = 1
    hY::T = 1
    hZ::T = 1
    origin::CartesianPoint{T} = zero(CartesianPoint{T})
    rotation::SMatrix{3,3,T,9} = one(SMatrix{3, 3, T, 9})
end
```
With this macro you can set up defaults very nicely. The macro will create some default constructor via keywords for you. 

PolygonOps.jl is a package for Polygon operations in 2D. They implemented special established algorithm. 
We use the `PolygonOps.inpolygon`-function which returns whether the point is outside, inside of the polygon or actually on the edge of it (in = 1, on = -1, out = 0).